engine/src/main/java/io/camunda/zeebe/streamprocessor/DirectProcessingResultBuilder.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.streamprocessor;
+
+import io.camunda.zeebe.engine.api.PostCommitTask;
+import io.camunda.zeebe.engine.api.ProcessingResult;
+import io.camunda.zeebe.engine.api.ProcessingResultBuilder;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.TypedResponseWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.TypedStreamWriter;
+import io.camunda.zeebe.msgpack.UnpackedObject;
+import io.camunda.zeebe.protocol.record.RecordType;
+import io.camunda.zeebe.protocol.record.RecordValue;
+import io.camunda.zeebe.protocol.record.RejectionType;
+import io.camunda.zeebe.protocol.record.ValueType;
+import io.camunda.zeebe.protocol.record.intent.Intent;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Implementation of {@code ProcessingResultBuilder} that uses direct access to the stream and to
+ * response writer. This implementation is here to support a bridge for legacy code. Legacy code can
+ * first be shaped into the interfaces defined in engine abstraction, and subseqeently the
+ * interfaces can be re-implemented to allow for buffered writing to stream and response writer
+ */
+final class DirectProcessingResultBuilder implements ProcessingResultBuilder {
+
+  private final List<PostCommitTask> postCommitTasks = new ArrayList<>();
+
+  private final StreamProcessorContext context;
+  private final TypedStreamWriter streamWriter;
+  private final TypedResponseWriter responseWriter;
+
+  private boolean hasResponse = false;
+
+  DirectProcessingResultBuilder(final StreamProcessorContext context) {
+    this.context = context;
+    streamWriter = context.getLogStreamWriter();
+    responseWriter = context.getTypedResponseWriter();
+  }
+
+  @Override
+  public ProcessingResultBuilder appendRecord(
+      final long key,
+      final RecordType type,
+      final Intent intent,
+      final RejectionType rejectionType,
+      final String rejectionReason,
+      final RecordValue value) {
+    streamWriter.appendRecord(key, type, intent, rejectionType, rejectionReason, value);
+    return this;
+  }
+
+  @Override
+  public ProcessingResultBuilder withResponse(
```

Comment:
That probably won't work yet. If you look at `TypedResponseWriterImpl` all those methods are mapped onto the same sort of structure:
```
  public void writeRejectionOnCommand(
      final TypedRecord<?> command, final RejectionType type, final String reason) {
    final byte[] bytes = reason.getBytes(StandardCharsets.UTF_8);
    stringWrapper.wrap(bytes);

    stage(
        RecordType.COMMAND_REJECTION,
        command.getIntent(),
        command.getKey(),
        type,
        stringWrapper,
        command.getValueType(),
        command.getRequestId(),
        command.getRequestStreamId(),
        command.getValue());
  }
```
I thought I had modeled the new interface based on the generic interface for all of them. Now, counting the parameters it is sex vs. nine, but then again some parameters are all derived from command. We probably also need to make the stage method public at some point, because this is the generic one. Just as this PR added `appendRecord(...)` to the interface. The method was already there, it just needed to be made accessible.

Anyways, I know it looks wild. I am sure we can make it work, though :-)